### PR TITLE
jenkins: add Jenkinsfile for CI pipeline

### PR DESCRIPTION
**Pull request by jieyu**
* run as a multibranch pipeline job in jenkins
* fix the build badge on README that was trying to access a protected URI

### CODE REVIEW

**Comment by jdef** [comment]
retest this please

**Comment by jdef** [comment]
retest this please

**Comment by jdef** [comment]
retest this please

**Comment by jdef** [comment]
retest this please

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    ])
+
+    checkout scm
+
+    stage("Prepare") {
+      sh("make rebuild-dev-image")
+    }
+
+    stage("Build and Test") {
+      sh("DOCKER=yes make check")
```
Comment (jdef) [inline Jenkinsfile]: maybe we could clean this up using `withEnv` to avoid expressing `DOCKER=yes` multiple times?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    stage("Verify author") {
+      def alerts_for_branches = params.ALERTS_FOR_BRANCHES.tokenize(",") as String[]
+      user_is_authorized(alerts_for_branches, params.SLACK_CREDENTIAL_ID, params.SLACK_CHANNEL)
+    }
+  }
+
+  node("mesos-ubuntu") {
+    properties([
+      parameters([
+        string(name: "GITHUB_CREDENTIAL_ID", defaultValue: "d146870f-03b0-4f6a-ab70-1d09757a51fc"),
```
Comment (jieyu) [inline Jenkinsfile]: Can we avoid checking in those default values in the repo?

Comment (jdef) [reply Jenkinsfile]: not really

**Pull request merged**